### PR TITLE
Adjust for move to gcc10 primary compiler

### PR DESCRIPTION
**Pull request by citrus-it**

### CODE REVIEW

**Comment by citrus-it** [comment]
@jclulow @rmustacc @richlowe - Please could you take a look?

File: docs/developers/build.md [hunk 1]
```diff
@@ -55,8 +55,7 @@ chosen:
 
 On OpenIndiana, the `build-essential` package includes the GCC compiler
 and other tools required to build illumos.  In addition, for now, we
-should install the newer GCC 7 compiler and the Python 3.5 runtime for a
-complete build.
+should install the newer GCC 10 compiler for a complete build.
```
[the remark this answers is not in the document]
Comment (citrus-it) [reply docs/developers/build.md:58]: I just checked and the OI build-essential contains gcc7 and 10, and python 3.9 but not 3.5, and the gate's Makefile.master still has 3.5 as the secondary python version. Updated.

[the remark this answers is not in the document]
Comment (citrus-it) [reply docs/developers/build.md:58]: Both are included in build-essential now

**Review by richlowe — APPROVED** [review]

**Comment by jclulow** [comment]
Thanks!  This has been deployed.

**Pull request merged**